intake/interface/source/defined_plots.py

```diff
@@ -128,15 +152,99 @@ def selected(self, selected):
         """When plot is selected set, make sure widget stays upto date"""
         self.select.value = selected
 
-    def callback(self, *events):
+    def plot_selected(self, *events):
         for event in events:
             if event.name == 'value':
                 self.pane.object = self._plot_object(event.new)
-            if event.name == 'options':
-                self.instructions.object = self.instructions_contents
+            self.custom.name = "Create" if str(self.select.value) == 'None' else "Edit"
+            self.edit_options.visible = self.custom.name == "Edit"
+
+    def name_changed(self, *events):
+        for event in events:
+            if event.name == "value_input":
+                # Empty name not allowed and name cannot already be in use
+                if len(event.new) == 0 or event.new in self.options:
+                    self.interact_save.disabled = True
+                else:
+                    self.interact_save.disabled = False
+
+    def cancel(self, _):
+        self.pane.object = self._plot_object(self.selected)
+        self.out[0] = self.pane
+        self.children[0].visible = True
+        self.children[1].visible = False
+        self._behavior_callback = None
+
+    def interact_action(self, _):
+        # Stop catalog from autoreloading
+        self.source.cat.ttl = None
+        # Perform action
+        self._behavior_callback()
+        # Save catalog (if YAMLFileCatalog)
+        if isinstance(self.source.cat, YAMLFileCatalog):
+            self.source.cat.add(self.source)
+        # Update description
+        self.callback_on_edit()
```

Comment:
I don't understand how this is supposed to be used. Surely at least `self` should be passed to the func, so that it can find out whatever information it needs.

Reply:
This is now a proper Event class with access to the Plot and various attributes describing which action was taken.